bin/stress-test/src/store_tests.rs

```diff
@@ -0,0 +1,213 @@
+        .map(request)
+        .buffer_unordered(concurrency)
+        .map(|res| res.unwrap())
+        .collect::<Vec<_>>()
+        .await;
+
+    print_summary(timers_accumulator);
+}
+
+/// Sends a single `sync_state` request to the store and returns a tuple with:
+/// - the elapsed time.
+/// - the response.
+pub async fn sync_state(
+    api_client: &mut ApiClient<InterceptedService<Channel, OtelInterceptor>>,
+    account_ids: Vec<AccountId>,
+    block_num: u32,
+) -> (Duration, SyncStateResponse) {
+    let note_tags = account_ids
+        .iter()
+        .map(|id| u32::from(NoteTag::from_account_id(*id, NoteExecutionMode::Local).unwrap()))
+        .collect::<Vec<_>>();
+
+    let account_ids = account_ids
+        .iter()
+        .map(|id| account_proto::AccountId { id: id.to_bytes() })
+        .collect::<Vec<_>>();
+
+    let sync_request = SyncStateRequest { block_num, note_tags, account_ids };
+
+    let start = Instant::now();
+    let response = api_client.sync_state(sync_request).await.unwrap();
+    (start.elapsed(), response.into_inner())
+}
+
+// SYNC NOTES
+// ================================================================================================
+
+/// Sends multiple `sync_notes` requests to the store and prints the performance.
+pub async fn bench_sync_notes(data_directory: PathBuf, iterations: usize, concurrency: usize) {
+    // load accounts from the dump file
+    let accounts_file = data_directory.join(ACCOUNTS_FILENAME);
+    let accounts = fs::read_to_string(accounts_file).await.unwrap();
+    let mut account_ids = accounts.lines().map(|a| AccountId::from_hex(a).unwrap()).cycle();
+
+    let store_client = start_store(data_directory).await;
+
+    // each request will have 3 note tags and will be sent with block number 0.
+    let request = |_| {
+        let mut client = store_client.clone();
+        let account_batch: Vec<AccountId> = account_ids.by_ref().take(3).collect();
+        tokio::spawn(async move { sync_notes(&mut client, account_batch).await })
+    };
+
+    // create a stream of tasks to send the requests
+    let timers_accumulator = stream::iter(0..iterations)
+        .map(request)
+        .buffer_unordered(concurrency)
+        .map(|res| res.unwrap())
+        .collect::<Vec<_>>()
+        .await;
+
+    print_summary(timers_accumulator);
+}
+
+/// Sends a single `sync_notes` request to the store and returns the elapsed time.
+pub async fn sync_notes(
+    api_client: &mut ApiClient<InterceptedService<Channel, OtelInterceptor>>,
+    account_ids: Vec<AccountId>,
+) -> Duration {
+    let note_tags = account_ids
+        .iter()
+        .map(|id| u32::from(NoteTag::from_account_id(*id, NoteExecutionMode::Local).unwrap()))
+        .collect::<Vec<_>>();
+    let sync_request = SyncNoteRequest { block_num: 0, note_tags };
+
+    let start = Instant::now();
+    api_client.sync_notes(sync_request).await.unwrap();
+    start.elapsed()
```

Comment:
I wonder if we want to introduce a couple of random note tags to account for non-standard use cases